client/Packages/com.beamable.server/Editor/Usam/Editor/Signposts/BeamServiceAsset.cs

```diff
@@ -7,5 +10,43 @@ namespace Beamable.Server.Editor.Usam
 	public class BeamServiceAsset : ScriptableObject
 	{
 		public BeamServiceSignpost data;
+
+		private void OnValidate()

```

Comment:
This is something that I'm not sure it's what we want. The ideal world would be that we would do this after we generate the csproj files of the assembly references, however this is state dependent, due to the use of the `DotnetService`, and the `SolutionPostProcessor` thing is static. I tried calling this before this part, but it yields an error:
```
AssemblyUtil.Reload();
CsharpProjectUtil.GenerateAllReferencedAssemblies();
```

Reply:
yeah the more I've worked with `OnValidate` functions, the more I hate them. They always seem to come and bite us in the butt due to AssetDatabase quirks on when they run. 

Can you remove this, and create a follow on Issue that describes how we need to be able to invoke this based on
1. the user "saving" their intention in the Inspector, 
2. when a sln file is modified.

Reply:
Just removed the `OnValidate` method, but I left the method that I made to validate the data in the asset. I think it will be useful for the saving the intention part.